src/Core/Resolvers/CosmosClientProvider.cs

```diff
@@ -49,38 +51,49 @@ private void InitializeClient(RuntimeConfig? configuration)
-            {
-                string userAgent = ProductInfo.GetDataApiBuilderUserAgent();
-                CosmosClientOptions options = new()
-                {
-                    ApplicationName = userAgent
-                };
+            IEnumerable<KeyValuePair<string, DataSource>> cosmosDb = configuration.DataSourceNameToDataSource.Where(x => x.Value.DatabaseType == DatabaseType.CosmosDB_NoSQL);
 
-                _connectionString = configuration.DataSource.ConnectionString;
-                ParseCosmosConnectionString();
-
-                if (!string.IsNullOrEmpty(_accountKey))
-                {
-                    Client = new CosmosClientBuilder(_connectionString).WithContentResponseOnWrite(true)
-                        .WithApplicationName(userAgent)
-                        .Build();
-                }
-                else if (string.IsNullOrEmpty(_accessToken))
-                {
-                    Client = new CosmosClient(_accountEndpoint, new DefaultAzureCredential(), options);
-                }
-                else
+            foreach (KeyValuePair<string, DataSource> dataSourcePair in cosmosDb)
+            {
+                string dataSourceName = dataSourcePair.Key;
+                DataSource dataSource = dataSourcePair.Value;
+                if (!Clients.ContainsKey(dataSourceName))
                 {
-                    TokenCredential servicePrincipal = new AADTokenCredential(_accessToken);
-                    Client = new CosmosClient(_accountEndpoint, servicePrincipal, options);
+                    CosmosClient client;
+                    string userAgent = ProductInfo.GetDataApiBuilderUserAgent();
+                    CosmosClientOptions options = new()
+                    {
+                        ApplicationName = userAgent
+                    };
+
+                    ParseCosmosConnectionString(dataSource.ConnectionString);
+
+                    if (!string.IsNullOrEmpty(_accountKey))
+                    {
+                        client = new CosmosClientBuilder(dataSource.ConnectionString).WithContentResponseOnWrite(true)
+                            .WithApplicationName(userAgent)
+                            .Build();
+                    }
+                    else if (_accessToken.ContainsKey(dataSourceName))
+                    {
+                        client = new CosmosClient(_accountEndpoint, new DefaultAzureCredential(), options);
+                    }
+                    else
+                    {
+                        TokenCredential servicePrincipal = new AADTokenCredential(_accessToken[dataSourceName]!);
+                        client = new CosmosClient(_accountEndpoint, servicePrincipal, options);
+                    }
+
+                    Clients.Add(dataSourceName, client);
                 }
             }
+
+            _defaultDataSourceName = configuration.DefaultDataSourceName;
```

Comment:
why do we need this private member? 
Also, why do we need the private members - _accountKey, _accountEndpoint?
Looking at code for multiple scenario, seems like at the end of this loop those private members will be set to the value of the last datasource. Is that what we want? Or do we need those values to be  default datasource members values?

Reply:
Removed default on class object. will be retrieved from runtime config like others. Fixed error related to accoundkey and accountendpoint pulling. We do not need those class objects now.